base/src/main/java/org/aya/core/def/CtorDef.java

```diff
@@ -40,17 +41,22 @@ public CtorDef(
   }
 
   /**
-   * @return first component: data's telescope, second component: con telescope
+   * @return first component: ctor's owner telescope, second component: ctor's self telescope
    */
   public static @NotNull DataDef.CtorTelescopes
   telescopes(@NotNull DefVar<CtorDef, TeleDecl.DataCtor> defVar) {
-    var core = defVar.core;
-    if (core != null) return new DataDef.CtorTelescopes(core.ownerTele, core.selfTele);
-    var dataSignature = defVar.concrete.patternTele;
-    assert dataSignature != null;
-    var conSignature = defVar.concrete.signature;
-    assert conSignature != null;
-    return new DataDef.CtorTelescopes(dataSignature, conSignature.param());
+    var concrete = defVar.concrete;
+    if (concrete != null) {
+      // See StmtTycker.java#L222
```

Comment:
Do not use a line number, because it changes over time without notice. You can use a function name, or just copy the code.

Reply:
I mean copy the code in the comment

Reply:
my bad